packages/app/src/App.tsx

```diff
@@ -163,41 +168,62 @@ function App(props: { flameFromQuery?: FlameFunction[] }) {
                   flameFunctions={flameFunctions}
                   renderInterval={finalRenderInterval()}
                   onExportImage={onExportImage()}
+                  edgeFade={showSidebar()}
                 />
               </WheelZoomCamera2D>
             </AutoCanvas>
           </div>
         </Root>
         <div class={ui.viewportControls}>
-          <div class={ui.buttonGroup}>
-            <button onClick={() => setZoom((p) => p * 0.9)}>
+          <ButtonGroup>
+            <For each={[1, 2, 4]}>
+              {(divider) => {
+                const pixelRatio_ = 1 / divider
+                return (
+                  <Button
+                    active={pixelRatio() === pixelRatio_}
+                    onClick={() => setPixelRatio(pixelRatio_)}
+                    style={{ 'min-width': '3rem' }}
+                  >
+                    <Show when={divider !== 1} fallback={'Full'}>
+                      <span>
+                        <sup>1</sup>/<sub>{divider}</sub>
+                      </span>
+                    </Show>
+                  </Button>
+                )
+              }}
+            </For>
+          </ButtonGroup>
+          <ButtonGroup>
+            <Button onClick={() => setZoom((p) => p * 0.9)}>
               <Minus />
-            </button>
-            <button class={ui.resetZoomButton} onClick={() => setZoom(1)}>
+            </Button>
+            <Button onClick={() => setZoom(1)} style={{ 'min-width': '4rem' }}>
               {(zoom() * 100).toFixed(0)}%
-            </button>
-            <button onClick={() => setZoom((p) => p / 0.9)}>
+            </Button>
+            <Button onClick={() => setZoom((p) => p / 0.9)}>
               <Plus />
-            </button>
-          </div>
-          <div class={ui.buttonGroup}>
-            <button
+            </Button>
+          </ButtonGroup>
+          <ButtonGroup>
+            <Button
               disabled={!history.hasUndo()}
               onClick={() => {
                 history.undo()
               }}
             >
               <Undo />
-            </button>
-            <button
+            </Button>
+            <Button
               disabled={!history.hasRedo()}
               onClick={() => {
                 history.redo()
               }}
             >
               <Redo />
-            </button>
-          </div>
+            </Button>
+          </ButtonGroup>
         </div>
         <Show when={showSidebar()}>
           <div class={ui.sidebar}>
```

Comment:
Not sure why I can comment only modified files, but below, delete flame button seems not rendered correctly?                    
![image](https://github.com/user-attachments/assets/996382ab-7d31-4d0c-8b8a-e7f5b6a64776)
Also, while we are at it, how about if we add three buttons like in desktop app windows, 
![image](https://github.com/user-attachments/assets/f1110c32-c68d-41a2-9c3d-eef16e034387)

Reply:
I'll take a look at the delete button issue, but the other one is unrelated to this change. Personally, I think we need the "add variation" button to be below the current variations at least, not inline with close button. Collapsing might not be needed when we switch to selecting transforms.

Reply:
Scrolling up issue is also unrelated, lets make it an issue.